### ISSUE TITLE
Spelling in two languages

### ISSUE THREAD

**Issue by robgal519** [valentjn/vscode-ltex#81]
Hi, I'm writing my master thesis, and I use a lot of English phrases, because of technical nature of my paper. 
Is there a way to include second dictionary? 

I do not really care for the grammar of the secondary language. I just get a lot of errors every time, I want to create a footnote with English translation of the term I was working on.
Because I do not need grammar this should make it easier for you to implement I hope.

The other possible solution would be to disable checking the dictionary, and I could then add third party tool to check for the spelling

Currently, I add all the words to my local dictionary in config, but I have to check every word before adding, and it is just annoying.

**Comment by valentjn** [comment]
Why not use [magic comments](https://valentjn.github.io/vscode-ltex/docs/advanced-features.html#magic-comments) together with [language-specific dictionaries](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexdictionary) (you can have two at the same time)? Also see [this question in the FAQ](https://valentjn.github.io/vscode-ltex/docs/faq.html#how-can-i-check-multiple-languages-at-once).

**Comment by robgal519** [comment]
Well, I could use magic comments, but it would add bloat to my text. Let's consider an example:
```
... dokumentacji możemy odnaleźć opis modułu FSMC (ang. Flexible Static Memory Controller), który służy do adresowania zewnętrznej pamięci. ...
```
It would need to become:
```
... dokumentacji możemy odnaleźć opis modułu FSMC 
% LTeX: language=en-US
(ang. Flexible Static Memory Controller),
% LTeX: language=pl-PL
 który służy do adresowania zewnętrznej pamięci. ...
```
This forced me to split the line, and add two commands around the foreign sentence. Where I just wanted to get rid of the squiggle line stating that it is not a valid word in Polish. If I could simply import English dictionary into Polish, it would solve my issue

**Comment by valentjn** [comment]
I see. But I'm not sure how the solution you propose would look like.

You say you want to "import English dictionary into Polish". So I assume you don't want to have spelling errors for any English words in your Polish document. But that would mean you could take a document that's completely in English, spell-check it while having the language set to Polish, and then there would be no spelling errors at all. That's clearly not desirable.

Actually, LanguageTool had a similar feature in versions 4.4 to 4.9.1, marked as experimental and called "alternative languages", see [this thread](https://forum.languagetool.org/t/first-step-towards-multi-lingual-lt/3420). However, it caused more problems than it solved, so [it was removed in LT 5.0](https://github.com/languagetool-org/languagetool/issues/2782#issuecomment-617610503).

Since you're using L<sup>A</sup>T<sub>E</sub>X, I think the proper way would be to use `babel` commands like `\foreignlanguage` to indicate the languages used, which you should do in any case as otherwise hyphenation of the English words will be wrong in your thesis. LT<sub>E</sub>X doesn't support these commands yet, and it's not quite easy to implement this. But it's possible. (If you wanted automatic multilingual support instead, that wouldn't be in LT<sub>E</sub>X's scope, but in LanguageTool's.)

In the meantime, since you said it's annoying to add words to the dictionary, remember that you can add multiple words at once. Just select text that contains the words and execute the `Add all unknown words in selection to dictionary` quickfix (`Ctrl+.`).

**Comment by robgal519** [comment]
Thank you for the comment, I did not expect that this would create such problems. I did not know that I need to use `\foreignlanguage`, I will definitely incorporate this to my thesis, and I will use magic comments to change language for ltex ( I created simple snippet, so it is not as much annoying)

**Comment by valentjn** [comment]
There's a reason why the 15-year old LanguageTool still doesn't really support multilingual documents; because it's not easy :wink:

I implemented support for babel commands for LT<sub>E</sub>X and it seems to work quite nicely. [Look here for a sneak preview of the syntax.](https://github.com/valentjn/vscode-ltex/blob/gh-pages-develop/pages/docs/advanced-features.md#multilingual-latex-documents-with-the-babel-package) It's going to be in the upcoming feature release.

**Comment by valentjn** [comment]
Fix released in 6.1.0.

**Comment by ainar** [comment]
Can you please add support to the _csquotes_ package?
Like `\foreignblockquote` or `\foreigncblockquote`.
